File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        pointerSmall = 0
        pointerBig   = 0
        while pointerSmall<len(s) and pointerBig<len(t):
            if s[pointerSmall] == t[pointerBig]:
                pointerSmall+=1
                pointerBig+=1
            else:
                pointerBig+=1
        if pointerSmall == len(s):
            return True
        else:
            return False


        

File: 0392-is-subsequence/0392-is-subsequence.cpp
static const bool Booster = [](){
    #pragma GCC optimize("OFast")
    std::ios_base::sync_with_stdio(0);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int smallPointer = 0;
        int bigPointer = 0;
        while (smallPointer < s.size() && bigPointer < t.size()){
            if (s[smallPointer] == t[bigPointer]){
                smallPointer++;
                bigPointer++;
            }else{
                bigPointer++;
            }
        }
        if (smallPointer == s.size()){
            return true;
        }else{
            return false;
        }
    }
};